pkg/controllers/resources/nodes/fake_syncer.go

```diff
@@ -278,6 +280,8 @@ func filterOutPhysicalDaemonSets(pl *corev1.PodList) []corev1.Pod {
 	return podsNoDaemonSets
 }
 
-func getNodeHost(nodeName, currentNamespace string) string {
-	return nodeName + "." + translate.Suffix + "." + currentNamespace + "." + constants.NodeSuffix
+func GetNodeHost(nodeName, currentNamespace string) string {
+	hostname := strings.ReplaceAll(nodeName, ".", "-") + "." + translate.Suffix + "." + currentNamespace + "." + constants.NodeSuffix
+	klog.V(2).Infof("translating nodename %q into hostname: %q", nodeName, hostname)
```

Comment:
It's verbosity, but I checked and this is currently unreachable AFAIK since we don't expose klog verbosity.

I got it working just by adding the klog flagset (just the -v flag), I'll send a separate PR for that though.